pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -84,9 +129,48 @@ func (r *CloudEntitySelectorReconciler) SetupWithManager(mgr ctrl.Manager) error
 		}); err != nil {
 		return err
 	}
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&cloudv1alpha1.CloudEntitySelector{}).
-		Complete(r)
+
+	if err := ctrl.NewControllerManagedBy(mgr).For(&cloudv1alpha1.CloudEntitySelector{}).Complete(r); err != nil {
+		return err
+	}
+	return mgr.Add(r)
+}
+
+func (r *CloudEntitySelectorReconciler) Start(context.Context) error {
```

Comment:
Add a description on what function is supposed to do

Reply:
Done